agent/agent_test.go

```diff
@@ -26,7 +26,7 @@ import (
 )
 
 var (
-	mockCtx = mock.AnythingOfType("*context.emptyCtx")
+	mockCtx = mock.Anything
```

Comment:
Instead of this, what do you think about using a matcher function -
```suggestion
	mockCtx = mock.MatchedBy(func(ctx context.Context) bool { return true})
```